### PR TITLE
Added product variation thumbnails slider in product listing pages.

### PR DESCRIPTION
**Pull request by moritzbappert**
### Ticket
- [Product variations: thumbnails slider in product listing pages](https://app.asana.com/0/484879736575808/1146029343680883)

### Description
- Added a thumbnail slider to the WooCommerce product listing pages (e.g. category pages) to preview the main product image and the first image of each variation that is enabled via a backend setting in `WooCommerce > Gallerya`.

### CODE REVIEW

File: assets/scripts/script.js [hunk 1]
```diff
@@ -15,9 +16,15 @@
           watchCSS: true,
           arrowShape: arrowShape
         };
-        if (navigation == false || thumbnails.length > 0) {
-          sliderArgs['pageDots'] = false;
+        if (typeof pageDots != 'undefined') {
+          // Let the pageDots property be overriden by a data-attribute.
+          sliderArgs.pageDots = pageDots == true;
         }
+        else if (navigation == false || thumbnails.length > 0) {
```
[the remark this answers is not in the document]
Comment (moritzbappert) [reply assets/scripts/script.js]: This one can't use strict evaluation as the `navigation` value comes from a data attribute, which gives us a string if defined. '0' and '1' are used here.

File: assets/scripts/script.js [hunk 2]
```diff
@@ -15,9 +16,15 @@
           watchCSS: true,
           arrowShape: arrowShape
         };
-        if (navigation == false || thumbnails.length > 0) {
-          sliderArgs['pageDots'] = false;
+        if (typeof pageDots != 'undefined') {
+          // Let the pageDots property be overriden by a data-attribute.
+          sliderArgs.pageDots = pageDots == true;
```
[the remark this answers is not in the document]
Comment (moritzbappert) [reply assets/scripts/script.js]: This one can't use strict evaluation as the `pageDots` value comes from a data attribute, which gives us a string if defined. '0' and '1' are used here.

File: assets/scripts/script.js [hunk 3]
```diff
@@ -15,9 +16,15 @@
           watchCSS: true,
           arrowShape: arrowShape
         };
-        if (navigation == false || thumbnails.length > 0) {
-          sliderArgs['pageDots'] = false;
+        if (typeof pageDots != 'undefined') {
```
[the remark this answers is not in the document]
Comment (moritzbappert) [reply assets/scripts/script.js]: This one can't use strict comparison as the `pageDots` value comes from a data attribute, which gives us a string if defined. '0' and '1' are used here.

File: assets/scripts/script.js [hunk 4]
```diff
@@ -4,9 +4,10 @@
...
       $('.js-gallerya-slider').each(function(index, element) {
-        var navigation = $(this).closest('.gallerya--slider').attr('data-gallerya-navigation');
-        var thumbnails = $(this).closest('.gallerya--slider').find('.js-gallerya-thumbnail-slider');
-        var count = $(this).closest('.gallerya--slider').find('[data-gallerya-count]');
+        var navigation = $(this).closest('.gallerya--slider, .gallerya--product-variation-slider').attr('data-gallerya-navigation');
+        var pageDots = $(this).closest('.gallerya--slider, .gallerya--product-variation-slider').attr('data-gallerya-page-dots');
+        var thumbnails = $(this).closest('.gallerya--slider, .gallerya--product-variation-slider').find('.js-gallerya-thumbnail-slider');
+        var count = $(this).closest('.gallerya--slider, .gallerya--product-variation-slider').find('[data-gallerya-count]');
```
[the remark this answers is not in the document]
Comment (moritzbappert) [reply assets/scripts/script.js]: Good suggestions, I've implemented them!

**Comment by moritzbappert** [comment]
@juanlopez4691 Thanks, I've implemented your refactoring suggestions!

File: assets/scripts/script.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (juanlopez4691) [reply assets/scripts/script.js]: @moritzbappert I don't get why this is a reason to not use strict comparison. If `pageDots` is defined (value is '0' or '1'), its `typeof` will be `string`, so the result of `typeof pageDots !== 'undefined'` for sure will be `false`.

File: assets/scripts/script.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (juanlopez4691) [reply assets/scripts/script.js]: Ditto, ditto 😄

File: assets/scripts/script.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (juanlopez4691) [reply assets/scripts/script.js]: As said, I don't see why not strict comparison.

[the remark this answers is not in the document]
Comment (moritzbappert) [reply assets/scripts/script.js]: @juanlopez4691 I would need to use `navigation === '0'` then, ok?

[the remark this answers is not in the document]
Comment (juanlopez4691) [reply assets/scripts/script.js]: Ok, I see the point now, sorry. Just leave as it is. The comparison should work for any "falsy" value.

File: assets/scripts/script.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (moritzbappert) [reply assets/scripts/script.js]: @juanlopez4691 I could use `sliderArgs.pageDots = pageDots === '1';` here. Is this really better?

File: assets/scripts/script.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (moritzbappert) [reply assets/scripts/script.js]: Changed this one...

File: assets/scripts/script.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (moritzbappert) [reply assets/scripts/script.js]: Left this one unchanged.

**Pull request merged**